File: Recursion/Sum of digits/By iterative.cpp
// including header file
#include<bits/stdc++.h>

// namespace
using namespace std;

// main function
int main(){
	
	// n store digits
	long long n;
  int count = 0;
  cin>>n;
	
	int sum = 0;
	while (n != 0) {
        sum = sum + n % 10;
        n = n / 10;
  }
        
	cout<< sum;
	
}

File: Recursion/Sum of digits/By iterative.java
import java.util.*;

public class MyClass {

    public static void main(String args[]) {
        
        // n store digits
	      int n;
        int count = 0;
        Scanner s = new Scanner(System.in);
        
        n = s.nextInt();
        
        int sum = 0;
	      while (n != 0) {
            sum = sum + n % 10;
            n = n / 10;
        }
        
        // print the value of sum
		    System.out.print(sum);
    }
}
